opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -13,25 +13,41 @@
  */
 package opentracing;
 
+
 /**
- * Tracer is a simple, thin interface for Span creation.
+ * Tracer is a simple, thin interface for Span creation, and Span propagation into different transport formats.
  */
 public interface Tracer {
 
   /**
-   * Create, start, and return a new Span with the given `operationName`, all without specifying a
-   * parent Span that can be used to incorporate the newly-returned Span into an existing trace.
+   * Create, start, and return a new Span with the given `operationName`.
+   * An optional parent Span can be specified used to incorporate the newly-returned Span into an existing trace.
    *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
-   * Span feed = tracer.startTrace("GetFeed");
-   * Span http = tracer.startTrace("HandleHTTPRequest")
+   * Span feed = tracer.startTrace("GetFeed", null);
+   * Span http = tracer.startTrace("HandleHTTPRequest", feed)
    *                   .setTag("user_agent", req.UserAgent)
    *                   .setTag("lucky_number", 42);
    * }</pre>
    */
-  Span startTrace(String operationName);
+  Span startSpan(String operationName, /* @Nullable */ Span parent);
+
+  /** Takes two arguments:
+   *    a Span instance, and
+   *    a “carrier” object in which to inject that Span for cross-process propagation.
+   *
+   */
+  <T> void inject(Span span, T carrier);
```

Comment:
Re the format parameter: I'm hoping we just don't need to add it to `opentracing-java` in the future. Since it's idiomatic to have an explosion of small facade classes in Java, I think that's realistic, too.

It had its merits, but having both `Tracer.getInjector(...)` and `Injector.injectSpan(...)` added too much cognitive load on the new OT user... my (very bright) coworker took a look at that scheme a few weeks ago and was quite confused, for example (that's actually why I opened https://github.com/opentracing/opentracing.github.io/issues/70). The transition to `Tracer.Inject()` in Go consolidated [these](https://github.com/opentracing/opentracing-go/pull/65/files#diff-66487bb56b848b3f1635aa3aa38f0f90L30) [three](https://github.com/opentracing/opentracing-go/pull/65/files#diff-66487bb56b848b3f1635aa3aa38f0f90L37) [comments](https://github.com/opentracing/opentracing-go/pull/65/files#diff-46db78603208f38452c0333bec12ab2cL30) into a single straight-line explanation of [Tracer.Inject()](https://github.com/opentracing/opentracing-go/pull/65/files#diff-46db78603208f38452c0333bec12ab2cR30)... much easier to grok, I think/hope, and that's a great thing for the fundamentally tricky (and essential) propagation API.